reana_server/reana_admin/cli.py

```diff
@@ -945,3 +955,94 @@ def check_workflows(
     else:
         click.secho("\nFAILED")
         sys.exit(1)
+
+
+@reana_admin.command()
+@click.option(
+    "--days",
+    "-d",
+    help="Close interactive sessions running longer than a specified number of days.",
+    required=True,
+    type=click.IntRange(min=0),
+)
+@click.option(
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    help="Show which interactive sessions would be closed. [default=False]",
+)
+def interactive_session_cleanup(days: int, dry_run: bool) -> None:
+    """Close prolonged interactive sessions."""
+    click.echo(
+        f"Starting to close interactive sessions running longer than {days} days.."
+    )
+    click.echo("Fetching interactive session pods..")
+    try:
+        pods = current_k8s_corev1_api_client.list_namespaced_pod(
+            namespace=REANA_RUNTIME_KUBERNETES_NAMESPACE,
+            label_selector="reana_workflow_mode=session",
+        ).items
+    except ApiException as e:
+        click.secho(f"Couldn't fetch a list of pods: {e}", fg="red", err=True)
+        sys.exit(1)
+
+    if not pods:
+        click.echo("There are no interactive sessions to process!")
+
+    for pod in pods:
+        try:
+            pod_name = pod.metadata.name
+            workflow_id = pod.metadata.labels["reana-run-session-workflow-uuid"]
+            user_id = pod.metadata.labels["reana-run-session-owner-id"]
+            container_args = pod.spec.containers[0].args
+            token = next(filter(lambda a: "token" in a, container_args)).split("'")[1]
+        except Exception as e:
+            click.secho(
+                f"Couldn't parse user details from '{pod_name}' session metadata: {e}",
+                fg="red",
+                err=True,
+            )
+            continue
+        host = REANA_HOSTNAME or "172.17.0.1:30443"
+        try:
+            session_status = requests.get(
+                f"https://{host}/{workflow_id}/api/status?token={token}", verify=False
```

Comment:
I don't really have a solution to this, but I don't like that we are making a lot of assumptions on how interactive sessions are deployed and configured (e.g. the token is passed as a command line argument, the status is accessible at `<workflow id>/api/status`...). This is somewhat made worse by the fact that some places of r-w-controller are ready to support multiple different session types, while here (and in other places of REANA) we assume that all the sessions are jupyter notebooks.

What is your opinion on having a new endpoint in r-w-controller to get the status of an interactive session? Given that we manage the deployment of the interactive sessions in r-w-controller, I suspect it might be easier to get the status of the session there, as all the configuration details should be known (e.g. port used by the service etc.)

Reply:
This would probably require some important changes to r-w-controller, though

Reply:
Created an issue to address this https://github.com/reanahub/reana-workflow-controller/issues/504